src/backend/distributed/commands/table.c

```diff
@@ -939,6 +939,32 @@ PreprocessAlterTableStmt(Node *node, const char *alterTableCommand,
 				 */
 				constraint->skip_validation = true;
 			}
+			else if (constraint->contype == CONSTR_PRIMARY)
+			{
+				if (constraint->conname == NULL)
+				{
+					Relation rel = RelationIdGetRelation(leftRelationId);
+					constraint->conname = ChooseIndexName(RelationGetRelationName(rel),
+														  RelationGetNamespace(rel),
+														  NULL, NULL, true, true);
+					RelationClose(rel);
+
+					Assert(GetMemoryChunkContext(constraint->conname) ==
```

Comment:
I believe this Assert is not needed. Also, let's not modify the original conname.